### PR TITLE
client: add a cli option for backward compatibility with non prefixed storage tries

### PR DESCRIPTION
**Pull request by g11tech**
Followup for 
 - https://github.com/ethereumjs/ethereumjs-monorepo/pull/3023

to enable running the client from previously unprefixed storage tries synced pre #3023

### CODE REVIEW

File: packages/client/bin/cli.ts [hunk 1]
```diff
@@ -328,6 +328,14 @@ const args: ClientOpts = yargs(hideBin(process.argv))
     describe: 'Force a snap sync run (for testing and development purposes)',
     boolean: true,
   })
+  .option('prefixStorageTrieKeys', {
+    describe:
+      'Enable/Disable storage trie prefixes (specify `false` for backward compatibility with previous states synced without prefixes)',
+    boolean: true,
+    default: true,
+    deprecated:
+      'Support for `--prefixStorageTrieKeys=false` is temporary. Please sync new instances with prefixStorageTrieKeys enabled',
```
Comment (scorbajio) [inline packages/client/bin/cli.ts:337]:
```suggestion
      'Support for `--prefixStorageTrieKeys=false` is temporary. Please sync new instances with `prefixStorageTrieKeys` enabled',
```

File: packages/client/src/config.ts [hunk 2]
```diff
@@ -60,6 +60,14 @@ export interface ConfigOptions {
    */
   forceSnapSync?: boolean
 
+  /**
+   * A temporary option to offer backward compatibility with state already synced
```
Comment (scorbajio) [inline packages/client/src/config.ts:64]:
```suggestion
   * A temporary option to offer backward compatibility with databases that are
```

File: packages/client/src/config.ts [hunk 3]
```diff
@@ -60,6 +60,14 @@ export interface ConfigOptions {
...
+   * with prefixed storage tries
```
Comment (scorbajio) [inline packages/client/src/config.ts:65]:
```suggestion
using non-prefixed keys for storage tries
```

**Comment by holgerd77** [comment]
Rebased this via UI

**Review by acolytec3 — APPROVED** [review]
LGTM

**Pull request merged**